crates/axum-utils/src/cookies.rs

```diff
@@ -101,7 +101,15 @@ impl CookieOption {
         cookie.set_http_only(true);
         cookie.set_secure(self.secure());
         cookie.set_path(self.path().to_owned());
-        cookie.set_same_site(SameSite::Lax);
+
+        // The `form_post` callback requires that, as it means a 3rd party origin will
+        // POST to MAS. This is presumably fine, as our forms are protected with a CSRF
+        // token
+        cookie.set_same_site(if self.secure() {
+            SameSite::None
+        } else {
+            SameSite::Lax
+        });
```

Comment:
confused about why this is gated on `secure()`.

Other than that, this feels a bit undesirable — `SameSite=Lax` is part of the reason why CSRF tokens are safe against some types of attacks.
 
As a not-very-thoroughly-thought-through suggestion: could we instead have the 3rd-party origin POST to some URL `/A`, which sanitises the POST and then performs a redirect (maybe this needs to be in JavaScript? not sure) to the intended destination `/B` to 'clean' it as a same-site request / bypass the `SameSite` restriction for only intended requests? (As I think the request to `/B` would then get the cookies, even though `/A` didn't...?? But I haven't tried this)